src/lib/circuit_value.ts

```diff
@@ -302,6 +302,14 @@ function circuitArray<T>(
       }
       return array.map((v) => elementType.toJSON(v));
     },
+    fromJSON(json) {
+      if (!('fromJSON' in elementType)) {
```

Comment:
why is this checked at runtime? To avoid introducing more layers of "ProvableExtendedExtended" or something? What are the cases in which the fromJSON would be missing?

Reply:
In `circuitArray`, I also accept a `Provable<T>` as input, without the 'extended'. The general idea was to accept every `Provable<T>` in most APIs that a user would interact with, and don't require methods like `fromJSON` on every type, to have an interface that is as light-weight as possible. Most use cases for provables don't involve the 'extended' stuff but just the basic to/fromFields method for using these types in a circuit

Reply:
👍